casper/src/main/scala/coop/rchain/casper/engine/Initializing.scala

```diff
@@ -70,7 +70,12 @@ class Initializing[F[_]: Sync: Metrics: Span: Concurrent: BlockStore: CommUtil:
                         casper <- MultiParentCasper
                                    .hashSetCasper[F](validatorId, genesis, shardId)
                         _ <- Engine
-                              .transitionToRunning[F](casper, approvedBlock, ().pure[F])
+                              .transitionToRunning[F](
+                                casper,
+                                approvedBlock,
+                                validatorId,
+                                ().pure[F]
```

Comment:
Nitpick - `F.unit` is a bit better

Reply:
I'm not a fan of `implicit F: Sync[F]`, b/c it's not always needed and requires more boilerplate. Since we have it, I could change that, but I think I'll skip it for the above reasons... Hope that's OK :)

Reply:
Yes - as I said, it's just a nitpick. Usually `Applicative.unit` does not allocate memory (an be val), so this can be "prematurely" faster. Looks like small price to pay. But, in the end it might not matter :-)